phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -776,8 +771,14 @@ private long validateAndGetServerTimestamp(TableRef tableRef, MultiRowMutationSt
         // If we're auto committing, we've already validated the schema when we got the ColumnResolver,
         // so no need to do it again here.
         PTable table = tableRef.getTable();
-        MetaDataMutationResult result = client.updateCache(table.getSchemaName().getString(), table.getTableName()
-                .getString());
+
+        // We generally don't re-resolve SYSTEM tables, but if it relies on ROW_TIMESTAMP, we must
```

Comment:
Would it make sense to change the check we have here for system tables to also check if the system table doesn't have ROW_TIMESTAMP column instead?
https://github.com/apache/phoenix/blob/dd5551b4e8cc17e5bd587e6c4f5f256a8212345a/phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java#L606

Reply:
@twdsilva I think that makes sense. I will update the PR. Thanks!